### PR TITLE
[DOC] Declare range API-stability (concatenated_sequences)

### PR DESCRIPTION
**Pull request by marehr**

### CODE REVIEW

File: include/seqan3/range/container/concatenated_sequences.hpp [hunk 1]
```diff
@@ -198,23 +221,32 @@ class concatenated_sequences
...
     static constexpr bool is_compatible_with_value_type = is_compatible_with_value_type_aux(std::type_identity<t>{});
 
-    //!\brief Whether a type is compatible with this class.
-    //!\hideinitializer
+    /*!\brief Whether a type is compatible with this class.
+     * \hideinitializer
+     * \details
+     * \noapi{Exposition only.}
+     */
     // cannot use the concept, because this class is not yet fully defined
```
Comment (Irallia) [inline include/seqan3/range/container/concatenated_sequences.hpp:238]: Why is this not a part of the comment block above?

File: include/seqan3/range/container/concatenated_sequences.hpp [hunk 2]
```diff
@@ -198,23 +221,32 @@ class concatenated_sequences
...
     static constexpr bool iter_value_t_is_compatible_with_value_type = true;
 
-    //!\brief Whether a type is compatible with this class.
-    //!\hideinitializer
+    /*!\brief Whether a type is compatible with this class.
+     * \hideinitializer
+     * \details
+     * \noapi{Exposition only.}
+     */
     // cannot use the concept, because this class is not yet fully defined
```
Comment (Irallia) [inline include/seqan3/range/container/concatenated_sequences.hpp:250]: same here

File: include/seqan3/range/container/concatenated_sequences.hpp [hunk 3]
```diff
@@ -198,23 +221,32 @@ class concatenated_sequences
...
     //!\endcond
 
-    //!\brief Whether a type is compatible with this class's `value_type` or `reference` type.
-    //!\hideinitializer
+    /*!\brief Whether a type is compatible with this class's `value_type` or `reference` type.
+     * \hideinitializer
+     * \details
+     * \noapi{Exposition only.}
+     */
     // we explicitly check same-ness, because these types may not be fully resolved, yet
```
Comment (Irallia) [inline include/seqan3/range/container/concatenated_sequences.hpp:229]: same here

**Review by Irallia — APPROVED** [review]
LGFM, just a tiny thing 💅

File: include/seqan3/range/container/concatenated_sequences.hpp (hunk 1 above)
Comment (marehr) [reply include/seqan3/range/container/concatenated_sequences.hpp:238]: This is more a message to the developer.

We could wrap that into `\if DEV`.

**Review by eseiler — APPROVED** [review]

**Pull request merged**